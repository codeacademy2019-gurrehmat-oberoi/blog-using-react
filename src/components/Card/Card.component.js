import React,{Component} from 'react';
import CardImage from '../CardImage/CardImage.component';
import CardBody from '../CardBody/CardBody.component';
import CardFooter from '../CardFooter/CardFooter.component';
import './Card.css'
const axios = require('axios');

class Card extends Component {
  state = {
    post: this.props.post
  };
  componentDidMount(){
    const externalDataPromise = axios.get('https://api.myjson.com/bins/hc5ye');
    externalDataPromise.then((response) => {
      this.setState({
        post: response.data[3],
      });
    });
  }
  render() {
    console.log(this.state.post);
    const post = this.state.post;
    return (
      <div className = 'Card' >
        <CardImage imageName= {post.image} />
        <CardBody 
          date= {post.date} 
          readTime = {post.readingTime}
          blogTitle = {post.title}
          blogPara = {post.description} 
        />
        <CardFooter
          claps = {post.claps}
          liked = {post.liked}
        />
        <button type='button' className='CreatePostButton' onClick={()=>(window.location.href = 'post')}>Create Post</button>
      </div>
    );
  }
}

Card.defaultProps = {
  post: {
    date: "2nd Januray, 2018",
    readingTime: "2 mins",
    title: "The future of abstract art and the culture ...",
    description: "Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...",
    claps: 10,
    liked: false,
    image: "abstract.png"
  }
}

export default Card;