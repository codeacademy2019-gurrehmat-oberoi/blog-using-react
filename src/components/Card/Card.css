.Card {
  /* padding: 5px; */
  margin: 3px;
  border: 3px solid;
  display: flex;
  flex-direction: column;
  align-items: center;
}