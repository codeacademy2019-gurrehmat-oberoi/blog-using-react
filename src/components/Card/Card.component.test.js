/**
* @jest-environment node
*/

import Card from './Card.component';
import React from 'react';
import renderer from 'react-test-renderer';
import axios from 'axios';

describe('Card', ()=> {
  it('should render a card', () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});