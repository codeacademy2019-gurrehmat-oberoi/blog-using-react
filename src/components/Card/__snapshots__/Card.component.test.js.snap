// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a card 1`] = `
<div
  className="Card"
>
  <img
    alt=""
    className="CardImage"
    src="abstract.png"
  />
  <div
    className="CardBody"
  >
    <div
      className="meta"
    >
      <span
        className="date"
      >
        2nd Januray, 2018
      </span>
      <span
        className="readTime"
      >
        2 mins
      </span>
    </div>
    <h1>
      The future of abstract art and the culture ...
    </h1>
    <p>
      Create a blog post subtitle that summarizes your post in a few short, punchy sentences and entices your...
    </p>
    <hr />
  </div>
  <div
    className="CardFooter"
  >
    <div
      className="left"
    >
      <img
        alt=""
        className="icon clap"
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "clapping.svg",
          }
        }
      />
      <span
        className="clapCount"
      >
         
        10
      </span>
    </div>
    <div
      className="right"
    >
      <img
        alt=""
        className="icon heart"
        src={
          Object {
            "ReactComponent": [Function],
            "__esModule": true,
            "default": "heart-black.svg",
          }
        }
      />
    </div>
  </div>
  <button
    className="CreatePostButton"
    onClick={[Function]}
    type="button"
  >
    Create Post
  </button>
</div>
`;
